api/api/admin/media_report.py

```diff
@@ -33,6 +38,20 @@ def _production_deferred(*values: str) -> Sequence[str]:
     return values
 
 
+def _non_production_deferred(*values: str) -> Sequence[str]:
+    """
+    Define a sequence in only the production environment.
+
+    The raw ID field is perfectly suited for massive tables, and so enabling
+    that in production will often prevent performance hits or outages. This will
+    return the input values only the environment is production, and in all other
+    cases it will return an empty sequence.
+    """
+    if settings.ENVIRONMENT != "production":
+        return ()
+    return values
```

Comment:
If would like to look at this separately as a follow up because this PR is pretty huge as it is and while this may remove a lot of code, which is good, it might make the diff huge and we also want to ensure we don't remove features that are useful or good to have in the admin.

Reply:
Issue filed https://github.com/WordPress/openverse/issues/4413.